test/tests/functional/pbs_hook_execjob_prologue.py

```diff
@@ -247,12 +250,21 @@ def test_prologue_hook_does_not_execute_twice_on_pbsdsh(self):
         attr = {'event': 'execjob_prologue'}
         self.server.create_import_hook(hook_name, attr, hook_body)
 
-        j = Job(TEST_USER, {'Resource_List.select': '2:ncpus=1',
-                            'Resource_List.place': 'scatter'})
-        j.create_script('pbsdsh -n 1 hostname')
+        j = Job(PBSROOT_USER, {'Resource_List.select': '2:ncpus=1',
+                               'Resource_List.place': 'scatter'})
+        j.create_script('#!/bin/sh\npbsdsh  hostname\nsleep 10\n')
         jid = self.server.submit(j)
-        self.server.expect(JOB, 'queue', op=UNSET, id=jid)
-        match = self.momB.log_match("Job;%s;executed prologue hook" % jid,
-                                    allmatch=True)
-        self.assertTrue(match is not None)
-        self.assertTrue(len(match) == 1)
+        attribs = self.server.status(JOB, id=jid)
+        self.server.expect(JOB, 'queue', op=UNSET, id=jid, offset=10)
+        host, opath = attribs[0]['Output_Path'].split(':', 1)
+        ret = self.du.cat(hostname=host, filename=opath)
+        _msg = "cat command failed with error:%s" % ret['err']
+        self.assertEqual(ret['rc'], 0, _msg)
+        mom1 = ret['out'][0].split(".")[0]
+        mom2 = ret['out'][1].split(".")[0]
+        self.exec_mom1 = self.moms[mom1]
+        self.exec_mom2 = self.moms[mom2]
+        self.exec_mom1.log_match("Job;%s;executed prologue hook" % jid,
+                                 allmatch=True)
+        self.exec_mom2.log_match("Job;%s;executed prologue hook" % jid,
+                                 allmatch=True)
```

Comment:
Since you are just checking for the existence of this message, why pass allmatch?  If you were doing something with the return value of log_match(), I could see it.  Since you aren't, I don't think it is necessary.